src/emulation_core/mips/datapath.rs

```diff
@@ -255,24 +269,44 @@ impl Datapath for MipsDatapath {
         self.registers[register]
     }
 
-    fn get_memory(&self) -> &Self::MemoryType {
+    fn set_register_by_str(&mut self, _register: &str, _data: Self::RegisterData) {
+        todo!()
+    }
+
+    fn initialize(&mut self, initial_pc: usize, instructions: Vec<u8>) -> Result<(), String> {
+        self.reset();
+        self.load_instructions(shims::convert_from_u8_bytes(instructions))?;
+        self.registers.pc = initial_pc as u64;
+        self.is_halted = false;
+
+        Ok(())
+    }
```

Comment:
I think I understand why Vec<u8> is being used here, to be architecture-agnostic, but since RISC-V and ARM instructions are also 32 bits, it might be easier to use Vec<u32> to reduce overhead and make it easier to store/fetch instructions. However, that's just my personal preference, and both approaches are equally valid

Reply:
I was thinking about this and I agree. I initially was worried about supporting architectures that aren't 32 bit, but not that many have instructions that are _above_ 32 bits, and if you have an instruction that's less than 32 bits you can still use a 32 bit int to store it. I'm going to revert this change.